uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/sub-controllers/toil-request-ctrl.js

```diff
@@ -0,0 +1,160 @@
+          vm.balance.change.amount = -vm.request.toil_to_accrue;
+          vm.balance.closing = vm.balance.opening + vm.balance.change.amount;
+          vm.uiOptions.showBalance = true;
+          vm.loading.calculateBalanceChange = false;
+        }
+      };
+
+      /**
+       * Calculates toil expiry date.
+       * TODO It will be based on from date for both single and multiple days for now.
+       *
+       * @return {Promise}
+       */
+      vm.calculateToilExpiryDate = function () {
+        if (!vm.request.from_date) {
+          vm.error = 'Please select from date to find expiry date';
+          return $q.reject(vm.error);
+        }
+
+        return AbsenceType.calculateToilExpiryDate(vm.request.type_id, vm.request.from_date)
+          .then(function (expiryDate) {
+            vm.expiryDate = expiryDate;
+          });
+      };
+
+      /**
+       * Checks if submit button can be enabled for user and returns true if successful
+       *
+       * @return {Boolean}
+       */
+      vm.canSubmit = function () {
+        return !!vm.request.duration && !!vm.request.toil_to_accrue &&
+          !!this.request.from_date && !!this.request.to_date;
+      };
+
+      /**
+       * This should be called whenever a date has been changed
+       * First it syncs `from` and `to` date, if it's in 'single day' mode
+       * Then, if all the dates are there, it gets the balance change
+       *
+       * @param {Date} date - the selected date
+       * @return {Promise}
+       */
+      vm.updateAbsencePeriodDatesTypes = function (date) {
+        var oldPeriodId = vm.period.id;
+
+        return vm._checkAndSetAbsencePeriod(date)
+          .then(function () {
+            var isInCurrentPeriod = oldPeriodId == vm.period.id;
+
+            if (!isInCurrentPeriod) {
+              if (vm.uiOptions.multipleDays) {
+                vm.uiOptions.showBalance = false;
+                vm.uiOptions.toDate = null;
+                vm.request.to_date = null;
+              }
+
+              return $q.all([
+                vm._loadAbsenceTypes(),
+                vm._loadCalendar()
+              ]);
+            }
+          })
+          .then(function () {
+            vm._setMinMaxDate();
+            vm._setDates();
+            vm.calculateToilExpiryDate();
+          })
+          .catch(function (error) {
+            vm.error = error;
+          });
+      };
+
+      /**
+       * Resets data in dates, types, balance in parent and reason here.
+       */
+      vm._reset = function () {
+        parentRequestCtrl._reset.call(this);
+        vm.request.toilDurationHours = 0;
+        vm.request.toilDurationMinutes = 0;
+        vm.request.updateDuration();
+        vm.request.toil_to_accrue = "";
+      };
+
+      /**
+       * Initializes the controller on loading the dialog
+       */
+      (function initController() {
+        vm.loading.absenceTypes = true;
+        initRequest();
+
+        vm.initParams = {
```

Comment:
Instead of explicitly creating one more variable we can use already existing `leaveType`
```
return AbsenceType.all({
 is_sick: this.leaveType === 'sick' ? true: false,
 allow_accruals_request: this.leaveType === 'toil' ? true: false
})
```
This way only changing the leaveType would be enough